src/Engine/Events/EventMap.h

```diff
@@ -2,16 +2,23 @@
 
 #include <unordered_map>
 #include <vector>
+#include <string>
 
 #include "Engine/Events/EventIR.h"
 
 class EventMap {
  public:
-    void add(int id, EventIR ir);
+    void add(int eventId, EventIR ir);
     void clear() { _eventsById.clear(); }
```

Comment:
What I would do is add a function like this one here:
`std::span<EventIR> events(int eventId)`

And use it in implementation instead of `valueOr`, because `valueOr` makes a copy.